lib/rex/shared_resource.rb

```diff
@@ -0,0 +1,9 @@
+# -*- coding: binary -*-
+
+# A tag interface used to indicate that the object in question should not be
+# replicated as part of the replicant process in modules.
+#
+# Note that atomic access isn't granted to the caller, and additional mutex/semaphores
+# may be required to behave as expected in a threaded context.
+module Rex::SharedResource
```

Comment:
This seems redundant with `Ref` or at least belongs in `rex-core`.
If you want to use the lookup as a determining factor, i'd suggest searching for `Ref` in the ancestors or one of the closer-down mixins like `Rex::Service` (basically an allow-list)

Reply:
The new namespace definitely has that advantage, agreed. It seems duplicative in its intent, but i do see the concern about "what else uses it" as being a pretty serious one. That said, the reason for doing this here is that anything already mixing in reference counters should (IMO) never be duplicated as the duplicated object's refcount would be a mess.